source/TextDatasource.js

```diff
@@ -163,6 +164,21 @@ class TextDatasource {
     toString() {
         return JSON.stringify(this.toObject());
     }
+
+    /**
+     * Creates a hash ID based on content
+     * @returns {String} using hash.js returns a hash of content
+     * @memberof TextDataSource
+     */
+    getID() {
+        if (this._content === "") {
+            throw new Error("this._content is empty");
```

Comment:
The consumer may not know of `this._content`.. It's just an implementation detail. Perhaps the error message could be a bit more descriptive, like so: "Failed getting ID: Datasource requires content for ID generation"